plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/TrinoCatalog.java

```diff
@@ -124,4 +125,8 @@ void createMaterializedView(
     void updateColumnComment(ConnectorSession session, SchemaTableName schemaTableName, ColumnIdentity columnIdentity, Optional<String> comment);
 
     Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName);
+
+    Map<String, Optional<Partition>> listPartitions(ConnectorSession session, SchemaTableName tableName);
+
+    io.trino.plugin.hive.metastore.Table getHiveTable(ConnectorSession session, SchemaTableName tableName);
```

Comment:
Rather than adding these here, could you inject a HiveMetastore instance into IcebergMetadata to get the Hive table information?

Reply:
Thanks, injected `HiveMetastoreFactory` to `MigrateProcedure` not to touch `IcebergMetadata`.